### ISSUE TITLE
Improve onEnd details

### ISSUE THREAD

**Issue by vitaly-t** [vitaly-t/iter-ops#161]
Operator [onEnd](https://vitaly-t.github.io/iter-ops/functions/onEnd.html) can be improved with extra details about duration, by providing min/max duration, along with indexes of where those were reached.

This is to help locate deviations in iterable performance.

For example, have `duration` inside `IIterationSummary` of the following type:

```ts
export interface IDuration {
    average: number; // averge duration
    max: {delay, index, value} // delay = waiting time, index when the value was reached, the value
    min: {delay, index, value}; // same as above
    total: number; // total duration
}
```

This would be a breaking change, because `duration` property changes.

**Comment by vitaly-t** [comment]
This change now has been merged into `master`.

**Comment by vitaly-t** [comment]
Updated [onEnd](https://vitaly-t.github.io/iter-ops/functions/onEnd.html) released in v2.0.0-beta.7
